controllers/secretutils/secret-utils.go

```diff
@@ -0,0 +1,158 @@
+package secretutils
+
+import (
+	"crypto/sha512"
+	"errors"
+	"fmt"
+	"strings"
+
+	multiclusterv1alpha1 "github.com/red-hat-storage/odf-multicluster-orchestrator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type SecretLabelType string
+
+const (
+	SourceLabel           SecretLabelType = "BLUE"
+	DestinationLabel      SecretLabelType = "GREEN"
+	InternalLabel         SecretLabelType = "INTERNAL"
+	IgnoreLabel           SecretLabelType = "IGNORE"
+	SecretLabelTypeKey                    = "multicluster.odf.openshift.io/secret-type"
```

Comment:
let's not group different types under the same `const` umbrella. either it should be separated by an empty line or it should be a new cost block